spec/integration/type/file_spec.rb

```diff
@@ -1528,4 +1528,161 @@ def pretty_transaction_error(transaction)
 
     raise "Got #{status_failures.length} failure(s) while applying: #{status_fail_msg}"
   end
+
+  describe "copying a file that is a link to a file" do
+    let(:target) { tmpfile('target') }
+    let(:link) { tmpfile('link') }
+    let(:copy) { tmpfile('copy') }
+    it "should copy the target of the link if :links => follow" do
+      catalog.add_resource described_class.new(
+        :name => target,
+        :ensure => "present",
+        :content => "Jenny I got your number / I need to make you mine")
+      catalog.add_resource described_class.new(
+        :name => link,
+        :ensure => "link",
+        :target => target)
+      catalog.add_resource described_class.new(
+        :name => copy,
+        :ensure => "present",
+        :source => link,
+        :links => "follow")
+      catalog.apply
+      expect(File.ftype(target)).to eq(File.ftype(copy))
+      expect(File.read(target)).to eq(File.read(copy))
+    end
+
+    it "should copy the link itself if :links => manage" do
+      catalog.add_resource described_class.new(
+        :name => target,
+        :ensure => "present",
+        :content => "Jenny I got your number / I need to make you mine")
+      catalog.add_resource described_class.new(
+        :name => link,
+        :ensure => "link",
+        :target => target)
+      # FIXME: do I need to clean up from the previous test?
+      catalog.add_resource described_class.new(
+        :name => copy,
+        :ensure => "present",
+        :source => link,
+        :links => "manage")
+      catalog.apply
+      expect(File.ftype(link)).to eq(File.ftype(copy))
+      expect(File.read(link)).to eq(File.read(copy))
+      expect(File.symlink?(copy))
```

Comment:
Does this assert that `symlink?` returns true (I'm not sure what the default behavior is when you don't specify `.to eq(true)`.

Also since it's a predicate method, if the expectation fails, it will just say false doesn't equal true. You can output better error messages using this form:

```
expect(File).to be_symlink(copy)
```